main.tf

```diff
@@ -469,6 +469,28 @@ data "aws_iam_policy_document" "bucket_policy" {
       }
     }
   }
+
+  dynamic "statement" {
+    for_each = length(var.source_ip_allow_list) > 0 ? [1] : []
+
+    content {
+      sid       = "AllowIPPrincipals"
+      effect    = "Deny"
+      actions   = ["s3:*"]
+      resources = [local.bucket_arn, "${local.bucket_arn}/*"]
+      principals {
```

Comment:
Is this section required?

Reply:
Sorry for not writing the description...
Yes, it is required to achieve what I wrote in the description now. 
Could you please review my PR again?

Reply:
Sorry what I meant was is the principals element required?

Reply:
Yes, it is required to be a valid policy. If we omit this we will get MalformedPolicyDocument error.

The reason for the wildcard is that I would like to block any principal to be blocked outside of specified IP.